src/encoded/types/cart.py

```diff
@@ -48,32 +54,29 @@ class Collection(Item.Collection):
         pass
 
 
-def _get_carts_by_user(request, userid):
-    request.datastore = 'database'
-    return [
-        request.resource_path(v, '')
-        for k, v in request.registry[COLLECTIONS]['cart'].items()
-        if v.properties['submitted_by'] == userid
-    ]
-
-
-def _create_cart(request, user):
+def _create_cart(request, user, name=None, identifier=None, status=None):
     carts = request.registry[COLLECTIONS]['cart']
     user_props = request.embed(request.resource_path(user), '@@object')
+    cart_name = name if name else '{} cart'.format(user_props['title'])
+    initial_cart = {
+        'submitted_by': str(user.uuid),
+        'status': status or 'current',
+        'name': cart_name,
+        'elements': []
+    }
+    if identifier:
+        initial_cart['identifier'] = identifier
     cart = create_item(
         carts.type_info,
         request,
-        {
-            'submitted_by': str(user.uuid),
-            'status': 'current',
-            'name': '{} cart'.format(user_props['title']),
-            'elements': []
-        }
+        initial_cart,
     )
-    return request.resource_path(cart)
+    cart_path = request.resource_path(cart)
+    cart.properties['current'] = cart_path
+    return cart_path
 
 
-@view_config(context=Cart.Collection, request_method='GET', permission='save-carts', name='get-cart')
+@view_config(context=Cart.Collection, request_method=('GET', 'PUT'), permission='save-carts', name='get-cart')
```

Comment:
It might be cleaner if we make a separate GET and PUT view.

Reply:
I separated GET and PUT views which I think addresses the other things. Thanks; that really makes things look better.